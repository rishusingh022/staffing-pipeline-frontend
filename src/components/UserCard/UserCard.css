.user-card-style img {
  height: 200px !important;
  object-fit: fill;
}
.bg-white {
  background-color: white;
}
