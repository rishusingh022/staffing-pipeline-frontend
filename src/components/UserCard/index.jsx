import React from 'react';
import Image from '../Image';
import './UserCard.css';
import { BsArrowRight } from 'react-icons/bs';
import PropTypes from 'prop-types';

export default function UserCard({
  imageUrl,
  altText,
  identityNumber,
  name,
  designation,
  location,
  handleButtonClick,
}) {
  return (
    <div className="shadow-lg flex flex-col w-56 user-card-style" data-testid="image-card">
      <Image imageUrl={imageUrl} altText={altText} />
      <div className="px-3 py-2">
        <div className="text-xs mb-1">FMNO: {identityNumber}</div>
        <div className="font-bold text-xl mb-1">{name}</div>
        <div className="font-semibold text-xs mb-1">{designation}</div>
        <div className="font-medium text-xs mb-1 text-gray-500">{location}</div>
      </div>
      <div className="link-button self-end text-xs mb-2 font-medium cursor-pointer" onClick={handleButtonClick}>
        Know More
        <BsArrowRight className="inline-block mx-2 text-blue-800" />
      </div>
    </div>
  );
}

UserCard.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
  identityNumber: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  designation: PropTypes.string.isRequired,
  location: PropTypes.string.isRequired,
  handleButtonClick: PropTypes.func.isRequired,
};
