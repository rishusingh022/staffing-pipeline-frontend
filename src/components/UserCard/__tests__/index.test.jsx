import React from 'react';
import { render } from '@testing-library/react';
import UserCard from '..';

describe('UserCard', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<UserCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
