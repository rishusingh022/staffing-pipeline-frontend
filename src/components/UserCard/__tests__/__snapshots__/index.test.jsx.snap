// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserCard should render correctly 1`] = `
<DocumentFragment>
  <div
    class="shadow-lg flex flex-col w-56 user-card-style"
    data-testid="image-card"
  >
    <img
      class="image-component"
    />
    <div
      class="px-3 py-2"
    >
      <div
        class="text-xs mb-1"
      >
        FMNO: 
      </div>
      <div
        class="font-bold text-xl mb-1"
      />
      <div
        class="font-semibold text-xs mb-1"
      />
      <div
        class="font-medium text-xs mb-1 text-gray-500"
      />
    </div>
    <div
      class="link-button self-end text-xs mb-2 font-medium cursor-pointer"
    >
      Know More
      <svg
        class="inline-block mx-2 text-blue-800"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 16 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M1 8a.5.5 0 0 1 .5-.5h11.793l-3.147-3.146a.5.5 0 0 1 .708-.708l4 4a.5.5 0 0 1 0 .708l-4 4a.5.5 0 0 1-.708-.708L13.293 8.5H1.5A.5.5 0 0 1 1 8z"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </div>
</DocumentFragment>
`;
